apps/remote_control/lib/lexical/remote_control/code_intelligence/docs.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Docs do
+  @moduledoc """
+  Utilities for fetching documentation for a compiled module.
+  """
+
+  alias Lexical.RemoteControl.CodeIntelligence.Docs.Entry
+
+  defstruct [:module, :doc, functions_and_macros: [], callbacks: [], types: []]
+
+  @type t :: %__MODULE__{
+          module: module(),
+          doc: Entry.content(),
+          functions_and_macros: %{optional(atom()) => [Entry.t(:function | :macro)]},
+          callbacks: %{optional(atom()) => [Entry.t(:callback)]},
+          types: %{optional(atom()) => [Entry.t(:type)]}
+        }
+
+  @doc """
+  Fetches known documentation for the given module.
+  """
+  @spec for_module(module()) :: {:ok, t} | {:error, any()}
+  def for_module(module) when is_atom(module) do
+    with :ok <- ensure_ready(module),
+         {:docs_v1, _anno, _lang, _fmt, module_doc, _meta, docs} <- Code.fetch_docs(module) do
+      {:ok, normalize_docs(module, module_doc, docs)}
+    end
+  end
+
+  defp normalize_docs(module, module_doc, element_docs) do
+    elements_by_kind = Enum.group_by(element_docs, &(&1 |> elem(0) |> elem(0)))
```

Comment:
this is a baffling function ;) What are the first elements of the tuples?

Reply:
Hah! I meant to add a comment, but I actually thing I should just be a little more verbose here in the interest of clarity.

```elixir 
elements_by_kind =
  Enum.group_by(element_docs, fn {{kind, _name, _arity}, ...} ->
    kind
  end)
```